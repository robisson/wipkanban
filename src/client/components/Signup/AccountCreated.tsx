import React from "react";
import {
  withStyles,
  Theme,
  StyleRulesCallback
} from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";
import green from "@material-ui/core/colors/green";

const styles: StyleRulesCallback = (theme: Theme) => ({
  root: {
    flexGrow: 1
  },
  paper: {
    padding: theme.spacing.unit * 2,
    height: "100vh",
    display: "flex",
    alignItems: "center"
  },
  backgroundPrimary: {
    backgroundColor: theme.palette.primary.main
  },
  text: {
    width: "100%"
  },
  link: {
    textDecoration: "none",
    color: theme.palette.secondary.main,
    "&:hover": {
      textDecoration: "underline"
    }
  },
  textSuccessfull: {
    color: green[500]
  }
});

interface IAccountCreatedProps {
  classes: any;
}

const AccountCreated: React.StatelessComponent<any> = ({
  classes
}: IAccountCreatedProps) => {
  return (
    <Grid container className={classes.root}>
      <Grid item xs={12} className={classes.demo}>
        <Grid
          container
          className={classes.demo}
          direction="row"
          justify="flex-start"
        >
          <Grid item xs={12} sm={12} md={7} lg={8} xl={8}>
            <Paper
              elevation={0}
              className={[classes.paper, classes.backgroundPrimary].join(" ")}
              style={{
                paddingTop: 10,
                paddingBottom: 10
              }}
            >
              <div>
                <br />
                <Typography
                  style={{
                    color: "white"
                  }}
                  variant="h4"
                  gutterBottom
                >
                  Welcome to the best open source plataform for lean management
                </Typography>
                <Typography
                  style={{
                    color: "white"
                  }}
                  variant="h5"
                  gutterBottom
                >
                  Align strategy and execution with <b>OKRs</b> and{" "}
                  <b>Kanban</b> for you, your team and your company.
                </Typography>
              </div>
            </Paper>
          </Grid>
          <Grid item xs={12} sm={12} md={5} lg={4} xl={4}>
            <Paper
              elevation={0}
              className={classes.paper}
              style={{
                paddingTop: 10,
                paddingBottom: 10
              }}
            >
              <div style={{ margin: "0 auto" }}>
                <Typography
                  className={classes.textSuccessfull}
                  align="center"
                  variant="h4"
                  gutterBottom
                >
                  Your account has been successfully created
                </Typography>
                <br />
                <Typography align="center">
                  Click at link in the bellow to do login
                </Typography>
                <br />
                <Typography variant="h5" align="center" gutterBottom>
                  <Link className={classes.link} to="/login">
                    Login
                  </Link>
                </Typography>
              </div>
            </Paper>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default withStyles(styles)(AccountCreated);
