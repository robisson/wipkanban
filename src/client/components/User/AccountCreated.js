import React from 'react';
import {Link} from 'react-router-dom';

const AccountCreated = () => {
    return (
        <div className="text-center">
            <h1>Your account has been successfully created.<br/> Click the link below to login</h1>
            <p>
                <Link to="/login" className="btn btn-success">
                    Login
                </Link>
            </p>
        </div>
    );
};

export default AccountCreated;