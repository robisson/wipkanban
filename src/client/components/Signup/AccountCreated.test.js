import React from "react";
import AccountCreated from "./AccountCreated";
import { createShallow } from "@material-ui/core/test-utils";
import { Link } from "react-router-dom";
import toJson from "enzyme-to-json";

// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<AccountCreated /> Component", () => {
  let shallow;
  let wrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<AccountCreated />);
  });

  it("should render component correctly", () => {
    expect(wrapper.type()).toEqual('div');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should have a h1 and a Link to /login", () => {
    expect(wrapper.find('h1').length).toEqual(1);
    expect(wrapper.find(Link).props().to).toEqual('/login')
  });
});
