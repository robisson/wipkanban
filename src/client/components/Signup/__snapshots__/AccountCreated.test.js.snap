// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountCreated /> Component should render component correctly 1`] = `
<div
  className="text-center"
>
  <h1>
    Your account has been successfully created.
    <br />
     Click the link below to login
  </h1>
  <p>
    <Link
      className="btn btn-success"
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </p>
</div>
`;
